### DISCUSSION TITLE
I have the same value for the variables on different devices.

### DISCUSSION THREAD

**Discussion by EpsilonOrionis** [remix-run/remix#9422] [Q&A]
Hello everyone,

I'm experimenting with fetcher to add and remove elements from an array. Everything works correctly, but when I access the app from another browser, I find the same values as the device on which I performed the initial array operations. The array and the methods are in a separate .ts file. I would like to know if this behavior of Remix is correct and how I can avoid sharing the state of variables between devices.

Thank you in advance.

**Comment by kiliman** [comment]
Is this array defined in a module? If so there is only one instance for the entire app. So all requests will share the same array. 

You can use Remix sessions to store data for individual users. 

https://remix.run/docs/en/main/utils/sessions

**Comment by EpsilonOrionis** [comment]
Thank you for your response. I have another problem with sessions that I can't figure out. Why is it only saved when I do a redirect, and I can't save it freely whenever I need to in my code?

**Reply by kiliman** [reply]
If you're using the cookie session storage, the entire session object is stored in a cookie. Anytime you mutate the session, you'll need to commit and add the `set-cookie` header:

```ts
const session = await getSession(request.headers.get('cookie'))
session.set('name', 'kiliman')
return json({ message: 'hello world', {
  headers: {
    'set-cookie': await commitSession(session),
  }
})
```

**Comment by EpsilonOrionis** [comment]
Thank you :)
